packages/typespec-ts/src/index.ts

```diff
@@ -219,6 +219,10 @@ export async function $onEmit(context: EmitContext) {
 
   async function generateRLCSources() {
     for (const rlcModels of rlcCodeModels) {
+      if (dpgContext.generationPathDetail?.rlcSourcesDir) {
+        rlcModels.options!.rlcSourceDir =
+          dpgContext.generationPathDetail?.rlcSourcesDir;
+      }
```

Comment:
Can you move this logic into https://github.com/Azure/autorest.typescript/blob/ff9e2dd7503b21ee80f8b1d9b9cd8b287bf6ef76/packages/typespec-ts/src/transform/transfromRLCOptions.ts#L40